Source/DafnyCore/CompileNestedMatch/MatchFlattener.cs

```diff
@@ -132,21 +96,21 @@ public MatchFlattener(ErrorReporter reporter, FreshIdGenerator idGenerator)
 
     CaseBody compiledMatch = CompilePatternPaths(state, new HoleCtx(), LinkedLists.Create(nestedMatchExpr.Source), paths);
     if (compiledMatch is null) {
-      // Happens only if the match has no cases, create a Match with no cases as resolved expression and let ResolveMatchExpr handle it.
-      var result = new MatchExpr(nestedMatchExpr.tok, nestedMatchExpr.Source, new List<MatchCaseExpr>(), nestedMatchExpr.UsesOptionalBraces);
-      result.Type = nestedMatchExpr.Type;
-      FillMissingCases(result);
-      return result;
-    } else if (compiledMatch.Node is Expression expression) {
+      Reporter.Error(MessageSource.Resolver, nestedMatchExpr.Tok, "match has no cases");
+      var emptyMatch = new MatchExpr(nestedMatchExpr.tok, nestedMatchExpr.Source, new List<MatchCaseExpr>(), nestedMatchExpr.UsesOptionalBraces);
+      emptyMatch.Type = nestedMatchExpr.Type;
+      return emptyMatch;
```

Comment:
I would have preferred to use a `HavocExpression` here instead of an empty `MatchExpr`, but it doesn't exist. Created an issue: https://github.com/dafny-lang/dafny/issues/3528